server/pulp/server/controllers/importer.py

```diff
@@ -69,14 +69,18 @@ def set_importer(repo_id, importer_type_id, repo_plugin_config):
     :raises exceptions.InvalidValue: if the values passed to create the importer are invalid
     """
     repo = model.Repository.objects.get_repo_or_missing_resource(repo_id)
-    validate_importer_config(repo.repo_id, importer_type_id, repo_plugin_config)
+
+    if not plugin_api.is_valid_importer(importer_type_id):
```

Comment:
You didn't write the `plugin_api.is_valid_importer()`, but I'll write my suggestion here anyway. Overall rather than having a caller check the return code of something like `is_valid_importer()` consider having `is_valid_importer()` raise the `PulpCodedValidationException` itself, if it's not valid. This would be a much more Pythonic way of doing things and allow the Exception handling to work naturally. You didn't write this, and maybe now isn't the time to change that.